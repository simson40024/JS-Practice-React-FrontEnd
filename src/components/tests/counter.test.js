import Counter from "../share/counter";
import { createRoot } from "react-dom/client";
import { screen, fireEvent } from "@testing-library/react";
import { act } from "@testing-library/react";

let container = null;
beforeEach(() => {
  // подготавливаем DOM-элемент, куда будем рендерить
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // подчищаем после завершения
  // unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe("Counter", () => {
  it("should show counter", () => {
    const handleFunc = jest.fn();
    act(() => {
      createRoot(container).render(
        <Counter counter={2} handleClick={handleFunc} />
      );
    });

    expect(container).toMatchSnapshot();
  });

  it("should clicked button - with param = -1", () => {
    const handleFunc = jest.fn();
    act(() => {
      createRoot(container).render(
        <Counter counter={2} handleClick={handleFunc} />
      );
    });

    fireEvent.click(screen.getByRole("button", { name: "-" }));
    expect(handleFunc).toBeCalledWith(-1);
  });

  it("should clicked button + with param = 1", () => {
    const handleFunc = jest.fn();
    act(() => {
      createRoot(container).render(
        <Counter counter={2} handleClick={handleFunc} />
      );
    });

    fireEvent.click(screen.getByRole("button", { name: "+" }));
    expect(handleFunc).toBeCalledWith(1);
  });
});
