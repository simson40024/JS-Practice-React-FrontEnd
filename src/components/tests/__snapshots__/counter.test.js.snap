// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should show counter 1`] = `
<div>
  <div
    class="product-quantity"
  >
    <button
      role="button"
    >
      -
    </button>
    <input
      readonly=""
      type="text"
      value="2"
    />
    <button
      role="button"
    >
      +
    </button>
  </div>
</div>
`;
